src/middlewares/csrf_test.ts

```diff
@@ -0,0 +1,423 @@
+         * });
+         * </script>
+         */
+        const res = await handler(
+          new Request("http://localhost/form", {
+            method: "POST",
+            headers: {
+              "content-type": "multipart/form-data",
+              "origin": "http://example.com",
+            },
+            body: "name=fresh",
+          }),
+        );
+
+        expect(res.status).toBe(403);
+        expect(simplePostHandler.calls.length).toBe(0);
+      });
+
+      it('should be 403 for "text/plain" cross origin', async () => {
+        /*
+         * via http://example.com
+         *
+         * <form action="http://localhost/form" method="POST" enctype="text/plain">
+         *   <input name="name" value="fresh" />
+         * </form>
+         * or
+         * <script>
+         * fetch('http://localhost/form', {
+         *   method: 'POST',
+         *   headers: {
+         *     'content-type': 'text/plain',
+         *   },
+         *   body: 'name=fresh',
+         * });
+         * </script>
+         */
+        const res = await handler(
+          new Request("http://localhost/form", {
+            method: "POST",
+            headers: {
+              "content-type": "text/plain",
+              "origin": "http://example.com",
+            },
+            body: "name=fresh",
+          }),
+        );
+
+        expect(res.status).toBe(403);
+        expect(simplePostHandler.calls.length).toBe(0);
+      });
+
+      it("should be 403 if request has no origin header", async () => {
+        const res = await app.handler()(
+          new Request("http://localhost/form", {
+            method: "POST",
+            headers: {
+              "content-type": "application/x-www-form-urlencoded",
+            },
+            body: "name=fresh",
+          }),
+        );
+
+        expect(res.status).toBe(403);
+        expect(simplePostHandler.calls.length).toBe(0);
+      });
+
+      it("should be 200 for application/json", async () => {
+        /*
+         * via http://example.com
+         * Assume localhost allows cross origin POST
+         *
+         * <script>
+         * fetch('http://localhost/form', {
+         *   method: 'POST',
+         *   headers: {
+         *     'content-type': 'application/json',
+         *   },
+         *   body: JSON.stringify({ name: 'fresh' }),
+         * });
+         * </script>
+         */
+        const res = await handler(
+          new Request("http://localhost/form", {
+            method: "POST",
+            headers: {
+              "content-type": "application/json",
+              origin: "http://example.com",
+            },
+            body: JSON.stringify({ name: "fresh" }),
+          }),
+        );
+
+        expect(res.status).toBe(200);
+        expect(await res.text()).toBe("fresh");
+      });
+
+      it('should be 403 for "Application/x-www-form-urlencoded" cross origin', async () => {
+        const res = await app.handler()(
```

Comment:
Nit: Let's be consistent. Other tests re-use the `handler()` function

```suggestion
        const res = await handler(
```

Reply:
I overlooked that—thank you!